daemon/cluster/cluster.go

```diff
@@ -687,6 +687,19 @@ func (c *Cluster) CreateService(s types.ServiceSpec) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
+	currentService, err := getService(c.getRequestContext(), c.client, serviceID)
+	if err != nil {
+		return err
+	}
+	ctnr := currentService.Spec.Task.GetContainer()
+	if ctnr == nil {
```

Comment:
@aaronlehmann I've added a check here, before checking for `encodedAuth` below. Is this reasonable? Similarly for `UpdateService`.